content/starter-configs/tutorials/drupal-10.md

```diff
@@ -0,0 +1,191 @@
+---
+title: "Drupal 10"
+date: 2020-07-15T11:14:19-04:00
+weight: 1
+---
+
+Wondering how to configure Tugboat for a typical Drupal 10 repository? Every Drupal site tends to have slightly
+different requirements, so you may need to do more customizing, but this should get you started.
+
+The following documentation assumes you are using Composer to manage your Drupal 9 project (typically with either the
+`drupal/recommended-project` or the `drupal-composer/drupal-project` projects).
+
+## Configure Drupal
+
+A common practice for managing Drupal's `settings.php` is to leave sensitive information, such as database credentials,
+out of it and commit it to git. Then, the sensitive information is loaded from a `settings.local.php` file that exists
+only on the Drupal installation location.
+
+This pattern works very well with Tugboat. It lets you keep a Tugboat-specific set of configurations in your repository,
+where you can copy it into place with a
+[configuration file command](/setting-up-services/how-to-set-up-services/leverage-service-commands/).
+
+Add or uncomment the following at the end of `settings.php`
+
+```php
+if (file_exists($app_root . '/' . $site_path . '/settings.local.php')) {
+  include $app_root . '/' . $site_path . '/settings.local.php';
+}
+```
+
+Add a file to the git repository at `.tugboat/settings.local.php` with the following content:
+
+```php
+<?php
+$databases['default']['default'] = array (
+  'database' => 'tugboat',
+  'username' => 'tugboat',
+  'password' => 'tugboat',
+  'prefix' => '',
+  'host' => 'mysql',
+  'port' => '3306',
+  'namespace' => 'Drupal\\Core\\Database\\Driver\\mysql',
+  'driver' => 'mysql',
+);
+// Use the TUGBOAT_REPO_ID to generate a hash salt for Tugboat sites.
+$settings['hash_salt'] = hash('sha256', getenv('TUGBOAT_REPO_ID'));
+```
+
+## Configure Tugboat
+
+The Tugboat configuration is managed by a [YAML file](/setting-up-tugboat/create-a-tugboat-config-file/) at
+`.tugboat/config.yml` in the git repository. Here's a basic Drupal 9 configuration you can use as a starting point, with
+comments to explain what's going on:
+
+```yaml
+services:
+  # What to call the service hosting the site.
+  php:
+    # Use PHP 8.1 with Apache; this syntax pulls in the latest version of PHP 8
+    image: tugboatqa/php:8.1-apache
+
+    # Set this as the default service. This does a few things
+    #   1. Clones the git repository into the service container
+    #   2. Exposes port 80 to the Tugboat HTTP proxy
+    #   3. Routes requests to the preview URL to this service
+    default: true
+
+    # Wait until the mysql service is done building
+    depends: mysql
+
+    # A set of commands to run while building this service
+    commands:
+      # Commands that set up the basic preview infrastructure
+      init:
+        # Bump the memory limit to help Drupal along.
+        - echo "memory_limit = 10240M" >> /usr/local/etc/php/conf.d/my-php.ini
```

Comment:
Is this necessary? What commands fail without this change?
Also, wow, this bumps the memory limit to 10GB.

Reply:
I was having trouble with the site timing out, but I'm wondering if that was a coincidence with other things happening on the server.  Yeah, that should be 1 GB also.  I'll remove that and see what happens.

Reply:
_I must've put a decimal point in the wrong place or something. I always do that. I always mess up some mundane detail._